src/transaction/log_replication_atomic.cpp

```diff
@@ -383,9 +385,17 @@ namespace cublog
   {
     assert (m_locked_classes.count (trid) > 0);
 
-    for (auto classoid : m_locked_classes[trid])
+    if (m_locked_classes.size() == 1)
       {
-	lock_unlock_object_and_cleanup (&thread_entry, &classoid, oid_Root_class_oid, SCH_M_LOCK);
+	// If there are no more DDL statements to replicate, the intention locks should also be removed.
```

Comment:
Do we really need this? Does it harm anything? The replicator is the only owner of the IX_LOCK on Rootclass in PTS. It may even help to reduce the cost of acquiring the next lock acquisition since it can skip acquiring the IX_LOCK.
If it's released strictly when a transaction ends, I think it has a point, but we even don't.

Reply:
low probability; However, I guess that the read transaction with repeatable read isolation level or higher is blocked to escalation from IS to S on the Rootclass. I will look into the related code more to check the situation I tell can occur.

Reply:
> the fact that replicator, which holds no locks at all, acquires IX_LOCK on the root class struck me as conceptually odd. In my understanding, having IX_LOCK on the root class implies holding X_LOCK on one or more entities (class or instance) beneath it. Hence, I hadn't considered it from your perspective.

Sorry to flip-flop, but I think it makes more sense than mine conceptually. As a lock owner, the replicator releases the IX_LOCK when it has no X_LOCK on a class. I didn't consider it this way. Although it is different from things on ATS (multiple IX_LOCK for each trx -> one IX_LOCK for the replicator), I think it has its point. I'll leave it to your choice.